src/main/cpp/lang/js.cpp

```diff
@@ -214,6 +214,95 @@ std::size_t match_hashbang_comment(std::u8string_view s) noexcept
     return length;
 }
 
+std::size_t match_escape_sequence(std::u8string_view str) noexcept
+{
+    // https://262.ecma-international.org/15.0/index.html#prod-EscapeSequence
+    // All escape sequences must start with backslash.
+    if (str.empty() || str[0] != u8'\\') {
+        return 0;
+    }
+
+    // Need 1 more char after the backslash.
+    if (str.length() < 2) {
+        return 1;
+    }
+
+    switch (str[1]) {
+    case u8'\'':
+    case u8'"':
+    case u8'\\':
+    case u8'b':
+    case u8'f':
+    case u8'n':
+    case u8'r':
+    case u8't':
+    case u8'v':
+    case u8'0': // Null character.
+        return 2;
+    }
+
+    // \xHHH
+    if (str[1] == u8'x') {
```

Comment:
You could just continue the `switch` and make this `case u8'x'`.

I think I'll refactor it like that once merged anyway, but you can also do that in this PR.